Makefile

```diff
@@ -202,6 +202,17 @@ THIRD_PARTY_NOTICE_LICENSE_TYPE = microsoft-oss
 $(eval $(call CREATE_THIRD_PARTY_NOTICES_RULE,ThirdPartyNotices.txt,foundation,False,False))
 $(eval $(call CREATE_THIRD_PARTY_NOTICES_RULE,bin/$(CONFIGURATION)/lib/xamarin.android/ThirdPartyNotices.txt,$(THIRD_PARTY_NOTICE_LICENSE_TYPE),True,False))
 
+# Used by External XA Build
+EXTERNAL_XA_PATH=$(topdir)
+EXTERNAL_GIT_PATH=$(topdir)/external
+
+prepare-external-git-dependencies:
+	$(call MSBUILD_BINLOG,prep-external-tasks) build-tools/xa-prep-tasks/xa-prep-tasks.csproj
+	$(call MSBUILD_BINLOG,prep-external-checkout) build-tools/xa-prep-tasks/xa-prep-tasks.csproj \
+		/t:CheckoutExternalGitSources /p:ExternalSourceDependencyDirectory='$(EXTERNAL_GIT_PATH)'
```

Comment:
Do we need similar changes in `PrepareWindows.targets`?

So you could do `msbuild Xamarin.Android.sln /t:Prepare` in some way to setup monodroid?

Reply:
I'd prefer to keep Prepare and `prepare-external-git-dependencies` separate, however I think it _could_ make sense to add another target to `Xamarin.Android.sln` for this, so that you could run an external git setup in one command rather than two?

Reply:
I'm mainly wanting _a way_ to do it on Windows, since this is in a `Makefile`. (Our Windows builds basically just ignore make files)

Maybe a new `PrepareExternal` target that you can run on `Xamarin.Android.sln`?

Reply:
The windows way would be the same as described in make:
* `msbuild build-tools/xa-prep-tasks/xa-prep-tasks.csproj`
* `msbuild build-tools/xa-prep-tasks/xa-prep-tasks.csproj /t:CheckoutExternalGitSources`

The only reason `/p:ExternalSourceDependencyDirectory` is passed above is so that it can be overridden from make for local builds or otherwise.

We could wrap the two commands above in one new target instead as proposed though:
* `msbuild Xamarin.Android.sln /t:PrepareExternal`

Or I could document these the two extra `xa-prep-tasks` msbuild invocations when I work on documentation updates later today.

Reply:
Yeah, I think you could add a `PrepareExternal` target in `PrepareWindows.targets`.

It's probably easier than documenting those commands.